### PR TITLE
Add data cube logic 

### PR DESCRIPTION
**Pull request by 0aps**

### CODE REVIEW

**Review by bujosa — APPROVED** [review]
LGTM

File: backend/src/controllers/dataWarehouseController.js [hunk 1]
```diff
@@ -96,6 +100,123 @@ export const createDataWareHouseJob = function createDataWareHouseJob() {
...
+    }));
+  const months = Array.from(Array(lastMonthsAmount + 1).keys())
+    .slice(1)
+    .map(month => ({
+      keyword: `M${String(month).padStart(2, '0')}`,
+      date: moment()
+        .subtract(month, 'months')
+        .toDate()
+    }));
+  const dates = years.concat(months);
```
Comment (goritm) [inline backend/src/controllers/dataWarehouseController.js:149]: I'd abstract this to another function

File: backend/src/controllers/dataWarehouseController.js [hunk 2]
```diff
@@ -96,6 +100,123 @@ export const createDataWareHouseJob = function createDataWareHouseJob() {
...
+    const results = await dataCubeModel.find(filter);
+
+    res.json(results);
+  } catch (e) {
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).send(e.message);
+  }
+};
+
+export const createDataCube = async (req, res) => {
+  const lastYearAmount = 3;
```
Comment (goritm) [inline backend/src/controllers/dataWarehouseController.js:131]:
```suggestion
  const lastYearsAmount = 3;
```

File: backend/src/controllers/dataWarehouseController.js [hunk 3]
```diff
@@ -96,6 +100,123 @@ export const createDataWareHouseJob = function createDataWareHouseJob() {
   );
 };
 
+export const findDataCube = async (req, res, next) => {
+  try {
+    const { query } = req;
+
+    if (query.v && query.operator && !Operator[query.operator]) {
+      return next(new InvalidRequest('Las operaciones deben ser una de: ' + Object.keys(Operator).join(', ')));
```
Comment (goritm) [inline backend/src/controllers/dataWarehouseController.js]: inglish plis

File: backend/src/controllers/dataWarehouseController.js [hunk 4]
```diff
@@ -96,6 +100,123 @@ export const createDataWareHouseJob = function createDataWareHouseJob() {
...
+  const months = Array.from(Array(lastMonthsAmount + 1).keys())
+    .slice(1)
+    .map(month => ({
+      keyword: `M${String(month).padStart(2, '0')}`,
+      date: moment()
+        .subtract(month, 'months')
+        .toDate()
+    }));
+  const dates = years.concat(months);
+  const filter = dates
```
Comment (goritm) [inline backend/src/controllers/dataWarehouseController.js:150]: Looks great! Maybe we should move the filter too?

**Review by goritm — APPROVED** [review]
This looks pretty great, thank you for this! 🚀 
Left some minor comments, but I'll approve it in the meantime

File: backend/src/controllers/dataWarehouseController.js (hunk 1 above)
Comment (bujosa) [reply backend/src/controllers/dataWarehouseController.js:149]: rt

File: backend/src/controllers/dataWarehouseController.js (hunk 4 above)
Comment (bujosa) [reply backend/src/controllers/dataWarehouseController.js:150]: rt

**Review by bujosa — CHANGES_REQUESTED** [review]

**Pull request merged**